sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/VisualStudioCacheAccessor.java

```diff
@@ -65,27 +54,36 @@ static JsonNode readJsonFile(String path) throws IOException {
      * @return a Map containing VS Code user settings
      */
     public Map<String, String> getUserSettingsDetails() {
-        JsonNode userSettings = getUserSettings();
-        Map<String, String> details = new HashMap<>();
-
-        String tenant = null;
-        String cloud = "AzureCloud";
-
-        if (userSettings != null && !userSettings.isNull()) {
-            if (userSettings.has("azure.tenant")) {
-                tenant = userSettings.get("azure.tenant").asText();
-            }
-
-            if (userSettings.has("azure.cloud")) {
-                cloud = userSettings.get("azure.cloud").asText();
-            }
+        Map<String, String> details = null;
+        String json = null;
+        try {
+            json = new String(Files.readAllBytes(Paths.get(getSettingsPath())));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
         }
-
-        if (!CoreUtils.isNullOrEmpty(tenant)) {
-            details.put("tenant", tenant);
+        try (JsonReader jsonReader = JsonProviders.createReader(json)) {
+            details = jsonReader.readObject(reader -> {
```

Comment:
Might as well return in the try-catch block so we don't need to initialize result as null earlier